roadmap/state-sync-protocol.md

```diff
@@ -0,0 +1,198 @@
+  <dt>Events</dt>
+  <dd>Each Chunk in an Execution Result also contains a <a href="https://github.com/onflow/flow-go/blob/e73e19a87860f470053b6a8b624d3dcd43ad23bf/model/flow/chunk.go#L8">hash of the Events generated during execution</a>. Since we know the <a href="https://github.com/onflow/flow-go/blob/e73e19a87860f470053b6a8b624d3dcd43ad23bf/model/flow/event.go#L25-L29">grouping</a> and <a href="https://github.com/onflow/flow-go/blob/e73e19a87860f470053b6a8b624d3dcd43ad23bf/model/flow/event.go#L32">ordering</a> of Events within transactions as well as the <a href="https://github.com/onflow/flow-go/blob/e73e19a87860f470053b6a8b624d3dcd43ad23bf/model/flow/chunk.go#L13">number of transactions</a> within a Chunk, it should be possible to reproduce the Event Collection hash for each Chunk and check that it is correct.</dd>
+  <dt>Collections</dt>
+  <dd>A block payload contains a list of <a href="https://github.com/onflow/flow-go/blob/e73e19a87860f470053b6a8b624d3dcd43ad23bf/model/flow/payload.go#L9">Collection Guarantees</a>, each of which contains <a href="https://github.com/onflow/flow-go/blob/e73e19a87860f470053b6a8b624d3dcd43ad23bf/model/flow/collectionGuarantee.go#L12">the ID of the Collection being guaranteed</a>. Since each Collection ID is simply a hash of the transaction IDs in the Collection, the Collection IDs can be compared with the downloaded Collections themselves.</dd>
+  <dt>Transaction Results</dt>
+  <dd>Transaction Results currently do not have on-chain commitments, but these will be added in the future, allowing the downloaded Transaction Results to be verified.</dd>
+</dl>
+
+If an Access node finds that any of the downloaded objects do not have the correct hash, then the Execution node must have put an incorrect CID in their Execution Result and should be slashed.
+
+> There is a Verifier's Dilemma here which needs to be addressed. We will not go into the details of how to solve this here.
+
+#### Verification Nodes as Verifiers
+
+Since Verification nodes are each responsible for only a subset of the Chunks within a block, Execution nodes will need to compute Execution Data and CIDs for each individual Chunk. Once Verification nodes have the [Chunk Data Pack](https://github.com/onflow/flow-go/blob/e73e19a87860f470053b6a8b624d3dcd43ad23bf/model/flow/chunk.go#L40), they can reproduce the CIDs and check that they are correct.
+
+## Requirements and Rewards
+
+The Access node goes from being an optional node serving as a proxy, to being a relayer of state information. It should correspondingly offer SLAs on state re-transmissions and get rewarded for it.
```

Comment:
I'd say the access node always had some aspects of "relaying" info, just use to be more protocol state related (and some misc data like events and result strings)

Also, as mentioned earlier, I believe discussions of the "optionality" of access nodes is still under discussion, so not sure if we want to include the idea here.

Reply:
yes, I should clarify and put "being a relayer of execution state information" here